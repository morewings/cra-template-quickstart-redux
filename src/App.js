import React from 'react';

// components
import Sample from 'components/Sample/Sample.container';
import SampleWrapped from 'components/SampleWrapped';

const App = props => (
  <div className="App">
    <Sample />
    <SampleWrapped />
  </div>
);

export default App;
