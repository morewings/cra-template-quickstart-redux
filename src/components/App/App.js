import React from 'react';
import Counter from '../Counter';
import Random from '../Random';
import classes from './App.module.css';

const App = () => (
  <div className={classes.container}>
    <Counter />
    <Random />
  </div>
);

export default App;
