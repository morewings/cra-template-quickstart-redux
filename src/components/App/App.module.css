.container {
  display: flex;
  margin: 0 auto;
  width: 960px;
}
