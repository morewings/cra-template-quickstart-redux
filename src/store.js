import {createStore} from 'redux';
import rootReducer from './Redux/reducers';

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__() // eslint-disable-line
);

export default store;
