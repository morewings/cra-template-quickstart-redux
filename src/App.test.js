import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Provider} from 'react-redux';
import store from 'store';
import App from 'App';

it('renders without crashing', () => {
  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
