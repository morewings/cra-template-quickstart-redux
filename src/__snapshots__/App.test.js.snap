// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div
      className="App"
    >
      <Connect(Sample)>
        <Sample
          increment={0}
          sample={[Function]}
        >
          <div>
            <div>
              Hello from sample component!
            </div>
            <button
              onClick={[Function]}
              type="button"
            >
              Click me!
            </button>
            <div>
              You clicked: 
              0
               times.
            </div>
          </div>
        </Sample>
      </Connect(Sample)>
      <Wrapper>
        <Sample
          injectedProp="someStateOrInstanceMethod"
        >
          <div>
            Hello I am wrapped!
          </div>
        </Sample>
      </Wrapper>
    </div>
  </App>
</Provider>
`;
